pkg/kube/inject/webhook.go

```diff
@@ -733,7 +709,7 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 			typeMetadata.Kind = controllerRef.Kind
 
 			// heuristic for deployment detection
-			if typeMetadata.Kind == "ReplicaSet" && strings.HasSuffix(controllerRef.Name, pod.Labels["pod-template-hash"]) {
+			if typeMetadata.Kind == "ReplicaSet" && pod.Labels["pod-template-hash"] != "" && strings.HasSuffix(controllerRef.Name, pod.Labels["pod-template-hash"]) {
```

Comment:
bugfix: previously pods using ReplicaSet directly were marked as "Deployment" for owner